nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,297 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.epgstation;
+
+  username = config.users.users.epgstation.name;
+  groupname = config.users.users.epgstation.group;
+
+  settingsFmt = pkgs.formats.json {};
+  settingsTemplate = settingsFmt.generate "config.json" cfg.settings;
+  preStartScript = let
+    readPassword = f: if (f != null)
+      then "$(head -n1 \"${f}\")"
+      else "INSECURE_GO_CHECK_CONFIGURATION_NIX";
+  in
+  pkgs.writeScript "epgstation-prestart" ''
+    #!${pkgs.runtimeShell}
+
+    PASSWORD="${readPassword cfg.basicAuth.passwordFile}"
+    DB_PASSWORD="${readPassword cfg.database.passwordFile}"
+
+    # setup configuration
+    sed \
+      -e "s,@password@,$PASSWORD,g" \
+      -e "s,@dbPassword@,$DB_PASSWORD,g" \
+      ${settingsTemplate} > /etc/epgstation/config.json
+    chmod 440 /etc/epgstation/config.json
+    chown "${username}:${groupname}" /etc/epgstation/config.json
+
+    # XXX: Use password authentication, since mysqljs does not yet support auth_socket
```

Comment:
XXX?

Reply:
Just indicates that this needs attention, so that we can replace passwords with socket authentication just like other NixOS modules once we can. `FIXME` might also be an option, but it's probably not going to  be actionable any time soon and hence `XXX`.